document-portal/file-transfer.c

```diff
@@ -426,7 +432,7 @@ add_files (GDBusMethodInvocation *invocation,
           return;
         }
 
-      if (!validate_fd (fd, app_info, VALIDATE_FD_FILE_TYPE_REGULAR, &st_buf, &parent_st_buf, &path, &fd_is_writable, NULL) ||
+      if (!validate_fd (fd, app_info, VALIDATE_FD_FILE_TYPE_ANY, &st_buf, &parent_st_buf, &path, &fd_is_writable, NULL) ||
```

Comment:
Should we keep using REGULAR here for fds that are not referring to directories?

Reply:
ANY mean either DIR or REGULAR, checked against S_ISREG () or S_ISDIR (). So this does exactly what we want.